plenum/server/node.py

```diff
@@ -2411,13 +2422,109 @@ def doStaticValidation(self, request: Request):
+                "not required for ledger id {}"
+                .format(self, request.reqId, ledger_id)
+            )
+            return
+
+        config_req_handler = self.ledger_to_req_handler.get(CONFIG_LEDGER_ID)
+        if not config_req_handler:
+            raise LogicError("Config request handler is missed")
+
+        taa = None
+        taa_data = config_req_handler.get_taa_data()
+        if taa_data is not None:
+            taa, taa_seq_no, taa_txn_time = taa_data
+
+        if not taa:
+            logger.trace(
+                "{} TAA acceptance passed for request {}: "
+                "taa is empty or missed"
+                .format(self, request.reqId)
+            )
+            return
+
+        # TODO not (None or empty)  ???
+        taa_digest = config_req_handler.get_taa_digest()  # TODO test that digest match taa
+        if not taa_digest:  # TODO test
+            raise LogicError(
+                "Txn Author Agreement digest is missed in state for taa: version {}, seq_no {}, txn_time {}"
+                .format(taa[TXN_AUTHOR_AGREEMENT_VERSION], taa_seq_no, taa_txn_time)
+            )
+
+        # TODO INDY-2068
+        taa_aml = {}
+        # taa_aml = config_req_handler.get_taa_aml()
+        # if not taa_aml:
+        #    raise TaaAmlNotSetError(
+        #        "Txn Author Agreement acceptance mechanism list is not defined"
+        #    ) # TODO test
+
+        if not request.taaAcceptance:
+            raise InvalidClientTaaAcceptanceError(
+                request.identifier, request.reqId,
+                "Txn Author Agreement acceptance is required for ledger with id {}"
+                .format(ledger_id)
+            )
+
+        r_taa_a_digest = request.taaAcceptance[f.TAA_ACCEPTANCE_DIGEST.nm]
+        if r_taa_a_digest != taa_digest:
+            raise InvalidClientTaaAcceptanceError(
+                request.identifier, request.reqId,
+                "Txn Author Agreement acceptance digest is invalid or non-latest:"
+                " provided {}, expected {}"
+                .format(r_taa_a_digest, taa_digest)
+            )
+
+        r_taa_a_mech = request.taaAcceptance[f.TAA_ACCEPTANCE_MECHANISM.nm]
+        if (r_taa_a_mech not in taa_aml) and False:  # TODO INDY-2068
+            # TODO
+            #   - list might be quite long
+            #   - should we return AML in reject
+            raise InvalidClientTaaAcceptanceError(
+                request.identifier, request.reqId,
+                "Txn Author Agreement acceptance mechanism is inappropriate:"
+                " provided {}, expected one of {}"
+                .format(r_taa_a_mech, list(taa_aml))
+            )
+
+        r_taa_a_ts = request.taaAcceptance[f.TAA_ACCEPTANCE_TIME.nm]
+        ts_lowest = (
+            taa_txn_time -
+            self.config.TXN_AUTHOR_AGREEMENT_ACCEPANCE_TIME_BEFORE_TAA
+        )
+        ts_higest = (
+            self.now +
```

Comment:
It looks like we need to use the time of the primary's PrePrepare this request is present in instead of the local node's time.

Reply:
yes, updated